word_search.py

```diff
@@ -0,0 +1,102 @@
+'''
+Created on 9 June 2017
+
+@author: Eva
+'''
+from queue import Queue
+
+class WordSearch:
+    
+    def __init__(self):
+        return
+        
+    def words_found_in_grid(self, rowNo, columnNo, grid, dictionary):
+        """Uses bfs to find valid words from a grid according to a given dictionary 
+        
+            Args:
+                rowNo: number of rows in the grid
+                columnNo: number of columns in the grid
+                grid: two-dimensional array of characters
+                dictionary: a dictionary object that holds the valid words and prefices
+            
+            Returns:
+                A list of words that were found on the grid
+        """
+        self.__pathsTaken = []
+        self.__foundWords = []
+        self.__words_to_search = Queue()
```

Comment:
It is not immediately obvious what you put on the queue and what it means.

You could do one of:
* document in a comment what the queue contains
* create a class to represent this information